source/posix/pipe.c

```diff
@@ -276,10 +276,11 @@ int aws_pipe_read(struct aws_pipe_read_end *read_end, struct aws_byte_buf *dst_b
     ssize_t read_val = read(read_impl->handle.data.fd, dst_buffer->buffer + dst_buffer->len, num_bytes_to_read);
 
     if (read_val < 0) {
-        if (errno == EAGAIN || errno == EWOULDBLOCK) {
+        int errno_value = errno;
```

Comment:
trivial: suggest we be super duper obvious that this is to avoid bugs, vs just being a style choice
```suggestion
        int errno_value = errno; /* logging may reset error, so cache it */
```